contracts/governance/Governor.sol

```diff
@@ -0,0 +1,110 @@
+pragma solidity ^0.5.11;
+pragma experimental ABIEncoderV2;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+
+/**
+ * @title Governor
+ * @dev The Governor holds the rights to stage and execute contract calls i.e. changing Livepeer protocol parameters. 
+ */
+contract Governor {
+    
+    using SafeMath for uint256;
+
+    address public owner;
+
+    /// @dev mapping of updateHash (keccak256(update) => executeBlock (block.number + delay)
+    mapping(bytes32 => uint256) updates; 
+
+    event OwnershipTransferred(address indexed previousOwner, address indexed newOwner);
+
+    event UpdateStaged(Update update, uint256 delay);
+
+    event UpdateExecuted(Update update);
+
+    event UpdateCancelled(Update update);
+
+    struct Update {
+        address[] target;
+        uint256[] value;
+        bytes[] data;
+    }
+
+   /// @notice Throws if called by any account other than the owner.
+    modifier onlyOwner() {
+        require(msg.sender == owner, "unauthorized");
+        _;
+    }
+
+
+    /// @notice Throws if called by any account other than this contract.
+    /// @dev Forces the `stage/execute` path to be used to call functions with this modifier instead of directly.
+    modifier onlyThis() {
+        require(msg.sender == address(this), "unauthorized");
+        _;
+    }
+
+
+    /// @dev The Ownable constructor sets the original `owner` of the contract to the sender account.
+    constructor() public {
+        owner = msg.sender;
+    }
+
+
+    /// @notice Allows the current owner to transfer control of the contract to a newOwner.
+    /// @dev Can only be called through stage/execute, will revert if the caller is not this contract's address.
+    /// @param newOwner The address to transfer ownership to.
+    function transferOwnership(address newOwner) public onlyThis {
+        require(newOwner != address(0), "newOwner is zero");
+        emit OwnershipTransferred(owner, newOwner);
+        owner = newOwner;
+    }
+    
+    /// @notice Stage a batch of updates to be executed.
+    /// @dev Reverts if the 'msg.sender' is not the 'owner'
+    /// @param _update Update to be staged.
+    /// @param _delay (uint256) Delay (in number of blocks) for the update.
+    function stage(Update memory _update, uint256 _delay) public onlyOwner {
+        bytes32 updateHash = keccak256(abi.encode(_update));
+        updates[updateHash] = block.number.add(_delay);
```

Comment:
Since we don't check if the update is already staged, it would be possible for the same update to be staged multiple times which would just set the execute block farther in the future. 

Should we be more strict here and disallow staging an existing update by adding `require(updates[updateHash] == 0, ...)`?